ras_party/controllers/account_controller.py

```diff
@@ -153,31 +154,43 @@ def _add_enrolment_and_auth(business, business_id, case_id, party, session, surv
 @with_db_session
 def change_respondent_enrolment_status(payload, session):
     """
-    Change respondent enrolment status for business and survey
-    :param payload:
-    :param session:
-    :return:
+    Change respondent enrolment status for business and survey, takes params from a payload dict
+
+    :param payload: A dictionary holding the values for the respondent party id being modified
     """
-    change_flag = payload['change_flag']
-    business_id = payload['business_id']
-    survey_id = payload['survey_id']
     respondent_id = payload['respondent_id']
+
+    respondent = query_respondent_by_party_uuid(respondent_id, session)
+    if not respondent:
+        logger.info("Respondent does not exist", respondent_id=respondent_id)
+        raise NotFound("Respondent does not exist")
+
+    return _change_respondent_enrolment_status(respondent=respondent,
+                                               survey_id=payload['survey_id'],
+                                               business_id=payload['business_id'],
+                                               status=payload['change_flag'],
+                                               session=session)
+
+
+def _change_respondent_enrolment_status(respondent, survey_id, business_id, status, session):
+
     logger.info("Attempting to change respondent enrolment",
-                respondent_id=respondent_id,
+                respondent_id=respondent.party_uuid,
```

Comment:
Does this change definitely not mess up any other code?  I'm asking incase another bit of code calls this function with respondent_id being the uuid

Reply:
Only 2 calls made here , and in both party_uuid used .